### PR TITLE
Have LazyIri accept urls with empty query

### PR DESCRIPTION
**Pull request by eg-ryte**

### CODE REVIEW

File: src/test/scala/fr/davit/capturl/IriSpec.scala [hunk 1]
```diff
@@ -117,4 +119,20 @@ class IriSpec extends FlatSpec with Matchers {
 
   }
 
+  "LazyIri" should "parse at least everything that StrictIri parses" in {
+    val url = "http://example.com/path?query#fragment"
+    Iri(url, Lazy) shouldBe Iri(url, Strict)
+  }
```
Comment (sflip) [inline src/test/scala/fr/davit/capturl/IriSpec.scala]: Strictly speaking, this is not what you are testing. What's the exact value of this test?

Comment (eg-ryte) [reply src/test/scala/fr/davit/capturl/IriSpec.scala]: To have a base case where the lazy and the strict version show identical behaviour, but I agree that this test probably doesn't bring a lot of value.

File: src/test/scala/fr/davit/capturl/parsers/PathParserSpec.scala [hunk 2]
```diff
@@ -22,6 +22,10 @@ class PathParserSpec extends FlatSpec with Matchers {
     parse("directory/?query") shouldBe Path.Segment("directory", Slash())                                -> "?query"
     parse("/one//path?query") shouldBe Slash(Segment("one", Slash(Segment("", Slash(Segment("path")))))) -> "?query"
 
+    // path with empty query
+    parse("path?") shouldBe Segment("path")                                                              -> "?"
```
Comment (RustedBones) [inline src/test/scala/fr/davit/capturl/parsers/PathParserSpec.scala]: `PathParser` isn't used for the delimiter in lazy Iri

File: src/main/scala/fr/davit/capturl/parsers/IriParser.scala [hunk 3]
```diff
@@ -60,7 +60,7 @@ trait IriParser
   }
 
   def rawQuery: Rule1[Option[String]] = rule {
-    optional('?' ~ capture(oneOrMore(!'#' ~ ANY)))
+    optional('?' ~ capture(zeroOrMore(!'#' ~ ANY)))
```
Comment (RustedBones) [inline src/main/scala/fr/davit/capturl/parsers/IriParser.scala]: This should also be applied in the fragment

File: src/test/scala/fr/davit/capturl/parsers/PathParserSpec.scala (hunk 2 above)
Comment (eg-ryte) [reply src/test/scala/fr/davit/capturl/parsers/PathParserSpec.scala]: I'm aware - should I delete the test?

File: src/test/scala/fr/davit/capturl/IriSpec.scala (hunk 1 above)
Comment (eg-ryte) [reply src/test/scala/fr/davit/capturl/IriSpec.scala]: I deleted the test case

File: src/test/scala/fr/davit/capturl/parsers/PathParserSpec.scala (hunk 2 above)
Comment (eg-ryte) [reply src/test/scala/fr/davit/capturl/parsers/PathParserSpec.scala]: I deleted the test case.

File: src/test/scala/fr/davit/capturl/IriSpec.scala [hunk 4]
```diff
@@ -117,4 +119,21 @@ class IriSpec extends FlatSpec with Matchers {
...
+
+  it should "accept empty queries" in {
+    Iri("path?", Lazy) shouldBe Iri("path?", Strict)
+    Iri("path?#fragment", Lazy) shouldBe Iri("path?#fragment", Strict)
+  }
+
+  it should "accept empty fragment" in {
+    Iri("path#", Lazy) shouldBe Iri("path#", Strict)
+    Iri("path#", Lazy) shouldBe Iri("path#", Strict)
```
Comment (RustedBones) [inline src/test/scala/fr/davit/capturl/IriSpec.scala]: I think this is supposed to be with '?'

File: src/test/scala/fr/davit/capturl/IriSpec.scala [hunk 5]
```diff
@@ -117,4 +119,21 @@ class IriSpec extends FlatSpec with Matchers {
...
+    Iri("path?#", Lazy) shouldBe Iri("path?#", Strict)
```
Comment (RustedBones) [inline src/test/scala/fr/davit/capturl/IriSpec.scala]: can you add a non empty query here ?

**Pull request merged**